File: baselibrary/src/main/java/com/ju/baselibrary/utils/LoadingUtil.java
package com.ju.baselibrary.utils;

import android.app.Activity;
import android.app.Dialog;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.ju.baselibrary.R;

import java.lang.ref.WeakReference;


/**
 * loadingDialog
 */

public class LoadingUtil {

    public static WeakReference<Dialog> loadingDialog;

    public static void show(final Activity context, boolean flag, String str) {

        if (context == null)
            return;
        LayoutInflater inflater = LayoutInflater.from(context);
        View v = inflater.inflate(R.layout.view_loading_dialog, null);// 得到加载view
        RelativeLayout layout = (RelativeLayout) v.findViewById(R.id.dialog_view);// 加载布局
        TextView tvText = (TextView) v.findViewById(R.id.tv_text);
        tvText.setText(str);
        if (loadingDialog != null && loadingDialog.get().isShowing()) return;

        if (context.getParent() != null)
            loadingDialog = new WeakReference<>(new Dialog(context.getParent(), R.style.loading_dialog));// 创建自定义样式dialog
        else {
            loadingDialog = new WeakReference<>(new Dialog(context, R.style.loading_dialog));// 创建自定义样式dialog
        }

        loadingDialog.get().setCancelable(true);// 可以用“返回键”取消
        loadingDialog.get().setCanceledOnTouchOutside(false);
        loadingDialog.get().setContentView(layout, new LinearLayout.LayoutParams(
                LinearLayout.LayoutParams.WRAP_CONTENT,
                LinearLayout.LayoutParams.WRAP_CONTENT));// 设置布局
        if (!context.isFinishing())
            loadingDialog.get().show();
    }

    public static void show(Activity context, String str) {
        show(context, true, str);
    }

    public static void dismiss() {
        if (loadingDialog != null && loadingDialog.get().isShowing()) {
            loadingDialog.get().dismiss();
            loadingDialog = null;
        }
    }

}

File: baselib/src/main/java/com/hzy/baselib/util/LoadingUtil.kt
package com.hzy.baselib.util

import android.app.Activity
import android.app.Dialog
import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.fragment.app.FragmentActivity


import com.hzy.baselib.R

import java.lang.ref.WeakReference


/**
 * loadingDialog
 */

object LoadingUtil {

    var loadingDialog: WeakReference<Dialog>? = null

    fun show(context: Activity, flag: Boolean, str: String) {

        if (context == null)
            return
        val inflater = LayoutInflater.from(context)
        val v = inflater.inflate(R.layout.view_loading_dialog, null)// 得到加载view
        val layout = v.findViewById<View>(R.id.dialog_view) as RelativeLayout// 加载布局
        val tvText = v.findViewById<View>(R.id.tv_text) as TextView
        tvText.text = str
        if (loadingDialog != null) return

        loadingDialog = if (context.parent != null)
            WeakReference(Dialog(context.parent, R.style.loading_dialog))// 创建自定义样式dialog
        else {
            WeakReference(Dialog(context, R.style.loading_dialog))// 创建自定义样式dialog
        }

        loadingDialog?.get()?.setCancelable(true)// 可以用“返回键”取消
        loadingDialog?.get()?.setCanceledOnTouchOutside(false)
        loadingDialog?.get()?.setContentView(
            layout, LinearLayout.LayoutParams(
                LinearLayout.LayoutParams.WRAP_CONTENT,
                LinearLayout.LayoutParams.WRAP_CONTENT
            )
        )// 设置布局
        if (!context.isFinishing)
            loadingDialog?.get()?.show()
    }

    fun show(context: Activity, str: String) {
        show(context, true, str)
    }

    fun dismiss() {
        if (loadingDialog != null) {
            loadingDialog?.get()?.dismiss()
            loadingDialog = null
        }
    }
}
